### PR TITLE
SUBMARINE-384. Add EditorConfig metafile to reduce growth of the indentation mess

### PR DESCRIPTION
**Pull request by jiwq**
### What is this PR for?
To reduce growth of the indentation mess

### What type of PR is it?
[Improvement]

### What is the Jira issue?
https://issues.apache.org/jira/browse/SUBMARINE-384

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# EditorConfig is awesome: http://EditorConfig.org
+
+# top-most EditorConfig file
+root = true
+
+# newline ending every file; no trailing space
```
Comment (yuanzac) [inline .editorconfig]: If the comment is no trailing space. trim_trailing_whitespace should be true

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+[{Makefile,go.mod,go.sum,*.go}]
+indent_style = tab
+
+# 2 space indentation
+[*.{java,json,sh}]
+indent_style = space
+indent_size = 2
+
+# 4 space indentation
+[*.{py,R,xml,js}]
```
Comment (yuanzac) [inline .editorconfig]: Indentation size of pom.xml is 2

**Comment by pingsutw** [comment]
There is an EditConfig file in `submarine-workbench/workbench-web/.editorconfig`, Do we need to combine to this config (`./submarine/.editorconfig`)

File: .editorconfig (hunk 1 above)
Comment (jiwq) [reply .editorconfig]: Done.

**Comment by jiwq** [comment]
> With this EditorConfig file, what settings need to be set in the development tool (e.g., IDEA)?

High-frequency IDEs(IDEA/PyCharm/WebStorm/VisualStudio and so on) come bundled with native support for EditorConfig. So the user do not need extra work. More info see [here](https://editorconfig.org/).

File: .editorconfig (hunk 2 above)
Comment (jiwq) [reply .editorconfig]: Done.

**Comment by jiwq** [comment]
> There is an EditConfig file in `submarine-workbench/workbench-web/.editorconfig`, Do we need to combine to this config (`./submarine/.editorconfig`)

@pingsutw Thanks for your focus, I think it's not conflicted. On the contrary, the subproject should better own itself EditorConfig. Any thoughts?

**Comment by pingsutw** [comment]
Okay. If it's not conflicted, I thought we don't need to change it.
Thanks @jiwq. LGTM

**Pull request closed without merging**